### PR TITLE
Make it possible to use the build dir in cmake config mode

### PR DESCRIPTION
**Pull request by sloriot**

### CODE REVIEW

**Comment by nmellado** [comment]
Thanks for this PR.
However it seems that we've lost the transitive linking which allowed to get boost symbols when linking to `io`.
Sounds weird...

**Comment by sloriot** [comment]
My bad, I thought the `*Targets.cmake` were some left overs.

**Comment by sloriot** [comment]
My last commit should fix it in install mode but I don't really know the clean way to do it for the build directory.
In passing do you still need all the export files?

**Comment by sloriot** [comment]
Got it, will commit in a minute.

**Comment by sloriot** [comment]
Should be good now.

File: CMakeLists.txt [hunk 1]
```diff
@@ -256,15 +253,44 @@ write_basic_package_version_file(
...
 
 # Configure '<PROJECT-NAME>Config.cmake'
+# version to be installed
 # Use variables:
 #   * targets_export_name
 #   * PROJECT_NAME
+# Used by Config.cmake.in
+set(OpenGR_INSTALL_INC_DIR "${CMAKE_INSTALL_PREFIX}/include")
+set(OpenGR_INSTALL_LIB_DIR "${CMAKE_INSTALL_PREFIX}/lib")
+set (OpenGR_EXPORT_TARGET_DIR \${CMAKE_CURRENT_LIST_DIR})
```
Comment (nmellado) [inline CMakeLists.txt]: Great thanks.
Is the `\` a typo ?
Shouldn't it be written as:
```suggestion
set (OpenGR_EXPORT_TARGET_DIR "${CMAKE_CURRENT_LIST_DIR}")
```

Comment (sloriot) [reply CMakeLists.txt]: I added it so that the file generated is the same as before the PR (do no inline the variable).

**Pull request merged**